saleor/checkout/utils.py

```diff
@@ -149,13 +149,22 @@ def add_variants_to_checkout(checkout, variants, quantities, channel_slug):
         if not product_channel_listing or not product_channel_listing.is_published:
             raise ProductNotPublished()
 
-    # create checkout lines
-    lines = []
+    lines = {line.variant.pk: line for line in checkout.lines.all()}
```

Comment:
I believe that docs for this function don't describe current logic.

Reply:
Maybe?
```suggestion
    variant_ids_in_lines = {line.variant.pk: line for line in checkout.lines.all()}
```

Reply:
```suggestion
    lines = {line.variant_pk: line for line in checkout.lines.all()}
```